packages/page-staking/src/Performance/CurrentList.tsx

```diff
@@ -0,0 +1,200 @@
+  className?: string;
+  favorites: string[];
+  targets: SortedTargets;
+  toggleFavorite: (address: string) => void;
+  session?: number;
+  currentSessionCommittee: string[];
+  sessionValidatorBlockCountLookup: Record<string, number>;
+}
+
+type AccountExtend = [string, boolean];
+
+interface Filtered {
+  validators?: AccountExtend[];
+}
+
+function sortAccountByFavourites (accounts: string[], favorites: string[]): AccountExtend[] {
+  return accounts
+    .map((accountId): AccountExtend => [
+      accountId,
+      favorites.includes(accountId)
+    ])
+    .sort(([, isFavA]: AccountExtend, [, isFavB]: AccountExtend): number => {
+      return isFavA === isFavB ? 0 : (isFavA ? -1 : 1);
+    });
+}
+
+function getFiltered (targets: SortedTargets | undefined, favorites: string[], currentSessionCommittee: string[]): Filtered {
+  if (!targets) {
+    return {};
+  }
+
+  return {
+    validators: sortAccountByFavourites(currentSessionCommittee, favorites)
+  };
+}
+
+function mapValidators (infos: ValidatorInfo[]): Record<string, ValidatorInfo> {
+  const result: Record<string, ValidatorInfo> = {};
+
+  for (let i = 0; i < infos.length; i++) {
+    const info = infos[i];
+
+    result[info.key] = info;
+  }
+
+  return result;
+}
+
+type SessionIndexEntry = [{ args: [EraIndex] }, Option<u32>];
+
+interface CommitteeSize extends Struct {
+  nonReservedSeats: u32
+  reservedSeats: u32,
+}
+
+function CurrentList ({ className, currentSessionCommittee, favorites, session, sessionValidatorBlockCountLookup, targets, toggleFavorite }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const { api } = useApi();
+  const [nameFilter, setNameFilter] = useState<string>('');
+  const sessionInfo = useCall<DeriveSessionProgress>(api.derive.session.progress);
+  // to be used later on
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const committeeSize = useCall<CommitteeSize>(api.query.elections.committeeSize);
+  const erasStartSessionIndex = useCall<SessionIndexEntry[]>(api.query.staking.erasStartSessionIndex.entries);
+
+  // to be used later on
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const [era, eraFirstSession, eraLastSession]: [number | undefined, number | undefined, number | undefined] = useMemo(
+    () => {
+      if (!sessionInfo || !erasStartSessionIndex || !session) {
+        return [undefined, undefined, undefined];
+      }
+
+      const erasStartSessionIndexLookup: Record<number, number> = {};
+
+      erasStartSessionIndex.filter(([, values]) => values.isSome)
+        .forEach(([key, values]) => {
+          const eraIndex = key.args[0];
+
+          erasStartSessionIndexLookup[eraIndex.toNumber()] = values.unwrap().toNumber();
+        });
+
+      let currentEra = sessionInfo.activeEra.toNumber();
+      let currentEraSessionStart: number = erasStartSessionIndexLookup[currentEra];
+      let currentEraSessionEnd = null;
```

Comment:
when using `let`, there is no need to assign initial value when it is not known, consider `let currentEraSessionEnd;`